shap/plots/_beeswarm.py

```diff
@@ -134,88 +133,91 @@ def beeswarm(shap_values, max_display=10, order=Explanation.abs.mean(0),
-                color_bar_label="*" + color_bar_label
-            )
-
-        if max_display is None:
-            max_display = 7
-        else:
-            max_display = min(len(feature_names), max_display)
-
-        interaction_sort_inds = order#np.argsort(-np.abs(values.sum(1)).sum(0))
-
-        # get plotting limits
-        delta = 1.0 / (values.shape[1] ** 2)
-        slow = np.nanpercentile(values, delta)
-        shigh = np.nanpercentile(values, 100 - delta)
-        v = max(abs(slow), abs(shigh))
-        slow = -v
-        shigh = v
-
-        pl.figure(figsize=(1.5 * max_display + 1, 0.8 * max_display + 1))
-        pl.subplot(1, max_display, 1)
-        proj_values = values[:, interaction_sort_inds[0], interaction_sort_inds]
-        proj_values[:, 1:] *= 2  # because off diag effects are split in half
-        beeswarm(
-            proj_values, features[:, interaction_sort_inds] if features is not None else None,
-            feature_names=feature_names[interaction_sort_inds],
-            sort=False, show=False, color_bar=False,
-            plot_size=None,
-            max_display=max_display
-        )
-        pl.xlim((slow, shigh))
-        pl.xlabel("")
-        title_length_limit = 11
-        pl.title(shorten_text(feature_names[interaction_sort_inds[0]], title_length_limit))
-        for i in range(1, min(len(interaction_sort_inds), max_display)):
-            ind = interaction_sort_inds[i]
-            pl.subplot(1, max_display, i + 1)
-            proj_values = values[:, ind, interaction_sort_inds]
-            proj_values *= 2
-            proj_values[:, i] /= 2  # because only off diag effects are split in half
-            summary(
-                proj_values, features[:, interaction_sort_inds] if features is not None else None,
-                sort=False,
-                feature_names=["" for i in range(len(feature_names))],
-                show=False,
-                color_bar=False,
-                plot_size=None,
-                max_display=max_display
-            )
-            pl.xlim((slow, shigh))
-            pl.xlabel("")
-            if i == min(len(interaction_sort_inds), max_display) // 2:
-                pl.xlabel(labels['INTERACTION_VALUE'])
-            pl.title(shorten_text(feature_names[ind], title_length_limit))
-        pl.tight_layout(pad=0, w_pad=0, h_pad=0.0)
-        pl.subplots_adjust(hspace=0, wspace=0.1)
-        if show:
-            pl.show()
-        return
+    # FIXME: introduce beeswarm interaction values as a separate function `beeswarm_interaction()` (?)
+    #   In the meantime, users can use the `shap.summary_plot()` function.
+    #
+    # # plotting SHAP interaction values
+    # if len(values.shape) == 3:
+    #
+    #     if plot_type == "compact_dot":
+    #         new_values = values.reshape(values.shape[0], -1)
+    #         new_features = np.tile(features, (1, 1, features.shape[1])).reshape(features.shape[0], -1)
+    #
```

Comment:
Note that this `len(values.shape) == 3` can never be true at the moment, because there is already a check that the shape is equal to 2 above.

Thus why the undefined name errors never surfaced before.

I've chosen to comment this out to remind us that we should re-introduce this functionality eventually. But probably as a separate function...